### PR TITLE
Locks

### PR DESCRIPTION
**Pull request by UndeadNappist**
Unfinished lock system.

### CODE REVIEW

File: code/game/machinery/doors/door.dm [hunk 1]
```diff
@@ -80,7 +80,7 @@
 	return ..()
 
 /obj/machinery/door/CollidedWith(atom/movable/AM)
-	if(operating || (obj_flags & EMAGGED))
+	if(operating || (obj_flags & EMAGGED) || (locked))
```
Comment (LetterN) [inline code/game/machinery/doors/door.dm]: if(operating || (obj_flags & EMAGGED) || locked)

File: code/game/machinery/doors/door.dm [hunk 2]
```diff
@@ -147,7 +147,7 @@
 
 /obj/machinery/door/proc/try_to_activate_door(mob/user)
 	add_fingerprint(user)
-	if(operating || (obj_flags & EMAGGED))
+	if(operating || (obj_flags & EMAGGED) || (locked))
```
Comment (LetterN) [inline code/game/machinery/doors/door.dm]: if(operating || (obj_flags & EMAGGED) || locked)

File: code/game/objects/items/lock.dm [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	var/locked
+	var/lock_data
+	var/static/lock_uid = 1
+
+/obj/item/lock_construct/New() // Same system machines use for UID. Could probably add a global UID for everything if you wanted and use it for shenanigans, or simpler loading.
+	..()
+	lock_data = lock_uid++
+	desc = "A heavy-duty lock for doors. It has a [lock_data] engraved on it."
+
+/obj/item/lock_construct/attackby(var/obj/item/I, var/mob/user) // Blatantly borrowed from Baystation coders and modified for simplicity. Thanks for pointing me in that direction, Rhicora.
```
Comment (LetterN) [inline code/game/objects/items/lock.dm]: the var/ is not needed

File: code/game/objects/items/lock.dm [hunk 4]
```diff
@@ -0,0 +1,66 @@
+/obj/item/lock_construct
+	name = "lock"
+	icon = 'icons/obj/lock.dmi'
+	w_class = WEIGHT_CLASS_SMALL
+	var/locked
+	var/lock_data
+	var/static/lock_uid = 1
+
+/obj/item/lock_construct/New() // Same system machines use for UID. Could probably add a global UID for everything if you wanted and use it for shenanigans, or simpler loading.
```
Comment (LetterN) [inline code/game/objects/items/lock.dm]: dont use New() use Initialize()

File: code/game/objects/structures/mineral_doors.dm [hunk 5]
```diff
@@ -71,8 +74,15 @@
 		return
 	if(isliving(user))
 		var/mob/living/M = user
+		if((/obj/structure/barricade in src.loc) || (locked))
```
Comment (LetterN) [inline code/game/objects/structures/mineral_doors.dm]: if((/obj/structure/barricade in src.loc) || locked)

**Comment by UndeadNappist** [comment]
Do not merge, does not work properly as of yet.

File: code/game/machinery/doors/unpowered.dm [hunk 6]
```diff
@@ -1,20 +1,43 @@
...
 		return ..()
 
-/obj/machinery/door/unpowered/emag_act()
-	return
+/obj/machinery/door/unpowered/CollidedWith(atom/movable/AM)
+	..()
+	if(Lock)
+		if(Lock.check_key())
+			return
+..()
```
Comment (LetterN) [inline code/game/machinery/doors/unpowered.dm]: what does this do? it seems like a loose ..()

File: code/game/objects/items/lock.dm [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+	var/locked
+	var/lock_data
+	var/static/lock_uid = 1
+
+/obj/item/lock_construct/Initialize() // Same system machines use for UID. Could probably add a global UID for everything if you wanted and use it for shenanigans, or simpler loading.
+	..()
+	lock_data = lock_uid++
+	desc = "A heavy-duty lock for doors. It has a [lock_data] engraved on it."
+
+/obj/item/lock_construct/attackby(/obj/item/I, var/mob/user) // Blatantly borrowed from Baystation coders and modified for simplicity. Thanks for pointing me in that direction, Rhicora.
```
Comment (LetterN) [inline code/game/objects/items/lock.dm]: `/obj/item/lock_construct/attackby(obj/item/I, mob/user) ` should fix it

**Comment by AndrewL97** [comment]
@LetterN I am finalizing the changes on this one, will have the changes up hopefully tonight

File: code/game/machinery/doors/unpowered.dm (hunk 6 above)
Comment (AndrewL97) [reply code/game/machinery/doors/unpowered.dm]: This is mostly a placeholder for future changes

check_key() in the lock just ```return Lock.locked``` at the moment. In future I want things like traps/signallers to be attachable. So if the door is locked and you try to open it, it pings off the signaler or explodes the door, etc.

**Comment by UndeadNappist** [comment]
PR #212

**Pull request closed without merging**